### PR TITLE
Add an integer vector multiplication example

### PR DESCRIPTION
**Pull request by marbre**

### CODE REVIEW

**Comment by marbre** [comment]
@simon-camp and @lucas-camp, I would appreciate if could already take a look.

File: samples/simple_vec_mul/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+      "simple_mul_int_bytecode_module_vmvx_c.c"
+    COMMAND generate_embed_data ${_GEN_EMBED_ARGS}
+    DEPENDS generate_embed_data simple_mul_int_bytecode_module_vmvx.vmfb
+  )
+
+  add_library(simple_mul_int_bytecode_module_vmvx STATIC "")
+  target_sources(simple_mul_int_bytecode_module_vmvx
+    PRIVATE
+    simple_mul_int_bytecode_module_vmvx_c.c
+    simple_mul_int_bytecode_module_vmvx_c.h
```
Comment (lucas-camp) [inline samples/simple_vec_mul/CMakeLists.txt:43]:
```suggestion
      simple_mul_int_bytecode_module_vmvx_c.c
      simple_mul_int_bytecode_module_vmvx_c.h
```

File: samples/simple_vec_mul/simple_mul_int.c [hunk 2]
```diff
@@ -0,0 +1,192 @@
+// Copyright 2021 The IREE Authors
+// Copyright 2022 Fraunhofer-Gesellschaft zur Förderung der angewandten
```
Comment (lucas-camp) [inline samples/simple_vec_mul/simple_mul_int.c:2]:
```suggestion
// Copyright 2023 Fraunhofer-Gesellschaft zur Förderung der angewandten
```

**Review by lucas-camp — COMMENTED** [review]
Looks good. I have two minor suggestions. The sample runs on a `Nucleo F411RE`.

**Review by lucas-camp — APPROVED** [review]
LGTM. I tested both samples on the `Nucleo F411RE` successfully.

**Pull request merged**